packages/service/src/domain/commands/processors/playback-current-command.ts

```diff
@@ -0,0 +1,73 @@
+import { Command } from '@service/domain/commands/model/command';
+import { CommandProcessingResponse, CommandProcessingResponseBuilder } from '@service/domain/commands/model/command-processing-response';
+import { CommandParameters, CommandParametersBuilder, CommandProcessor } from '@service/domain/commands/model/command-processor';
+import { RegisterCommand } from '@service/domain/commands/registry/register-command';
+import { PlayerStateRequestEvent } from '@service/event-stream/model/events';
+import { PlayerEventStream } from '@service/event-stream/player-event-stream';
+import { PlayerState } from '@service/domain/player-state/player-state';
+import { Service } from 'typedi';
+import { Song } from '@service/lib';
+
+@RegisterCommand
+@Service()
+class PlaybackCurrentCommand extends CommandProcessor {
+  constructor(private playerEventStream: PlayerEventStream) {
+    super();
+  }
+
+  async execute(command: Command): Promise<CommandProcessingResponse> {
+    const includeQueue = command.rawArgs === 'queue';
+
+    const playerStateRequest: PlayerStateRequestEvent = { id: 'PlayerStateRequestEvent' };
+    const currentPlayerState: PlayerState = await this.playerEventStream.sendToPrimaryPlayer(playerStateRequest);
+
+    const text = this.buildMessage(
+      currentPlayerState.currentlyPlaying?.song ?? null,
+      currentPlayerState.queue ?? [],
+      includeQueue,
+    );
+
+    return new CommandProcessingResponseBuilder()
+      .fromMarkdown(text)
+      .build();
+  }
+
+  private buildMessage(current: Song | null, queue: Song[], includeQueue: Boolean): string {
+    const queueSongsTitles = includeQueue ? queue.map((song) => `  - ${song.name.replace(/(.{50})..+/, '$1…')}`) : [];
```

Comment:
Tak samo jak w poprzednim PRze - ten `$1` to działa w ogóle?

Reply:
Może warto wydzielić to do jakiejś utils funkcji? `truncate`? No i czy w sumie tego nie da się przypadkiem zrobić bez regexów tylko samym `String.substring`?

Reply:
Działa. Użyłem regexa bo był fajnym oneLinerem, ale faktycznie doszedł drugi PR z tym samym wykorzystaniem więc wyciągnięcie do utilsa i bardziej przyzwoita implementacja to dobry pomysł.